ddtrace/tracer/tracer.go

```diff
@@ -364,17 +368,11 @@ func (t *tracer) sample(span *span) {
 		return
 	}
 	if rs, ok := sampler.(RateSampler); ok && rs.Rate() < 1 {
-		// the span was sampled using a rate sampler which wasn't all permissive,
```

Comment:
This lock and check is not needed since `sample` is called only for new spans, where the only shared structure is the `trace` in the `context` which has its own locks.